www/docs/nextjs/ssg.md

```diff
@@ -14,41 +14,83 @@ Static site generation requires executing tRPC queries inside `getStaticProps` o
 ## Fetch data in `getStaticProps`
 
 ```tsx
-import Head from 'next/head';
-import { trpc } from '../utils/trpc';
+/**
+ * Given we're this file something like `pages/posts/[id].tsx`
+ */
 import { createSSGHelpers } from '@trpc/react/ssg';
+import {
+  GetStaticPaths,
+  GetStaticPropsContext,
+  InferGetStaticPropsType,
+} from 'next';
+import { prisma } from 'server/context';
+import { appRouter } from 'server/routers/_app';
+import superjson from 'superjson';
+import { trpc } from 'utils/trpc';
 
-// Statically fetch the data in getStaticProps
-export const getStaticProps = async (
-  context: GetStaticPropsContext<{ filter: string }>,
-) => {
-  const ssg = createSSGHelpers({
+
+export async function getStaticProps(
+  context: GetStaticPropsContext<{ id: string }>,
+) {
+  const ssg = await createSSGHelpers({
     router: appRouter,
-    transformer, // optional
     ctx: {},
+    transformer: superjson, // optional - adds superjson serialization
+  });
+  const id = context.params?.id as string;
+  await ssg.fetchQuery('post.byId', {
+    id,
   });
```

Comment:
I wonder if it's worth encouraging a kind of ad-hoc pattern here, or thinking about whether trpc could help in some way making sure what's statically fetched matches what's passed to `useQuery` later. e.g.

```ts
const postByIdQueryParams: trpc.SomeHelperType<'post.byId'> = params => ['post.byId', params]

//...

export const getStaticProps = async context => {
  //...
  await ssg.fetchQuery(...postByIdQueryParams(context.params))
  //...
}

export default function PostViewPage(props) {
  const postQuery = trpc.useQuery(postByIdQueryParams(props))
  //...
}
```

Having written that... it's a bit much for this PR.